mobile/src/screens/meditation/meditation.tsx

```diff
@@ -13,15 +14,27 @@ type RouteParameters = {
 };
 
 const Meditation: React.FC = () => {
-  const { selectedMeditationEntry } = useAppSelector(({ meditation }) => {
-    return {
-      selectedMeditationEntry: meditation.selectedMeditationEntry,
-    };
-  });
+  const { selectedMeditationEntry, meditationEntries } = useAppSelector(
+    ({ meditation }) => {
+      return {
+        selectedMeditationEntry: meditation.selectedMeditationEntry,
+        meditationEntries: meditation.meditationEntries,
+      };
+    },
+  );
 
   const route = useAppRoute();
   const { duration }: RouteParameters = route.params as RouteParameters;
 
+  useEffect(() => {
+    if (selectedMeditationEntry) {
+      const currentTrackIndex = meditationEntries.findIndex((item) => {
+        return item.id === selectedMeditationEntry.id;
+      });
+      void player.setFirstTrack(Number(currentTrackIndex));
```

Comment:
we don't need number conversion, but we need to handle case where track is not found
```suggestion
      void player.setFirstTrack(currentTrackIndex);
```

Reply:
We already have `if ` to check if selectedMeditationEntry exists. If it doesn't, the first track in the queue will be set as the default first track